lib/vernier.rb

```diff
@@ -31,9 +31,44 @@ def self.profile(mode: :wall, out: nil, gc: true, **collector_options)
     result
   end
 
+  @collector = nil
+  @collector_out = nil
+
+  def self.start_profile(mode: :wall, out: nil, gc: true, **collector_options)
+    if @collector
+      @collector.stop
+      @collector = @collector_out = nil
+
+      raise "Profile already started, stopping..."
+    end
+
+    gc &&= (mode == :retained)
+    3.times { GC.start } if gc
```

Comment:
(non-blocking) I'd also like to avoid this duplication (since it's a bit strange to do anyways and I should probably adjust it to only GC once 😅). Maybe it can also be moved into the collector.

Reply:
I agree, will address in a follow up together with the collector out.

Reply:
https://github.com/jhawthorn/vernier/pull/61